system/index.js

```diff
@@ -20,10 +20,33 @@ const userFn = getFunction(USER_FUNCTION_URI);
 module.exports = async ({headers, payload}) => {
   if (DEBUG) {
     console.log('==System Function Start==');
-    console.log('HEADERS:', headers);
-    console.log('PAYLOAD:', payload);
+    console.log(`HEADERS: ${JSON.stringify(headers)}`);
+    console.log(`ORIGINAL PAYLOAD: ${payload}`);
+    console.log(`MIDDLEWARE_FUNCTION_URI: ${MIDDLEWARE_FUNCTION_URI}`);
+    console.log('==Middleware Function(s) Start==');
+  }
+
+  if (MIDDLEWARE_FUNCTION_URI) {
+    await Promise.all(MIDDLEWARE_FUNCTION_URI.split(':').map(async (middleware) => {
+          try {
+            const middlewareFn = await getFunction(middleware);
```

Comment:
It looks like we're going to lookup/load these middlewares for each function invocation. Could we build the list of middlewares once, and only call them during invocation?

Reply:
I think to do that, the node-function-buildpack would have to make sure it's the last buildpack to be run. We'd then have to have this env var set as a build var and could then hardcode that list to disk/env var that is read at runtime.

Reply:
I was thinking something like what we're already doing for loading the user function: https://github.com/heroku/node-function-buildpack/pull/24/files#diff-5c350865b90e8c9c7f2b316f9c84f74fL18. 

Alternatively, we could do this in `$init`. That might get weird since we'd potentially still have to call the customer's `$init`.

Reply:
ah, I see what you mean now. :+1:

Reply:
👍  I'll look into it, @joshwlewis